modules/reindex/src/main/java/org/elasticsearch/index/reindex/ParentBulkByScrollTask.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.reindex;
+
+import org.elasticsearch.tasks.TaskId;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Task for parent bulk by scroll requests that have sub-workers.
+ */
+class ParentBulkByScrollTask extends BulkByScrollTask {
+    /**
+     * Workers are built after this task is started but in the same thread that starts this task. 
+     */
+    private final BulkByScrollTask[] workers;
+
+    public ParentBulkByScrollTask(long id, String type, String action, String description, TaskId parentTaskId, int workerCount) {
+        super(id, type, action, description, parentTaskId);
+        workers = new BulkByScrollTask[workerCount];
+    }
+
+    /**
+     * Record one of the workers.
+     */
+    void setWorker(int index, BulkByScrollTask worker) {
+        assert workers[index] == null;
+        workers[index] = worker;
```

Comment:
I feel like this duplicates that information that is already available in the task manager (all child tasks of a particular task) and it is possible to perform this function ether directly against task manager (if we want to limit everything to a single node) or using TransportTasksAction if we want to make it generic.